### ISSUE TITLE
List Type

### ISSUE THREAD

**Issue by lesar** [funkia/list#4]
Hello, I liked to know what kind of Lists you have implemented: if your List have a math name or are
inspired to a math algorithm?

regards

**Comment by paldepind** [comment]
Good question. I should probably document that somewhere.

It's an implementation of relaxed radix balanced trees (RRB-trees).

By far the easiest to understand introduction to the data structure is [this series](http://www.hypirion.com/musings/understanding-persistent-vector-pt-1) of blog posts. The blog posts describes a simpler version of the data structure. I'm sure if it has a specific name but most often it's simply called "persistent vector".

List implements an extension to the persistent vector called relaxed radix balanced search trees. They were invented in [this paper](https://infoscience.epfl.ch/record/169879/files/RMTrees.pdf) (the paper isn't that easy to understand though). I've found that [this thesis](http://hypirion.com/thesis.pdf) is the best explanation of RRB-trees. [This thesis](https://infoscience.epfl.ch/record/205070/files/main.pdf) also has some additional implementation techniques. I've also added a few ideas on my own in the implementation to speed things up further 😄
